### ISSUE TITLE
AI: add crud operations for a conversations table to the AI service 

### ISSUE THREAD

**Issue by chirino** [trustification/trustify#1082]
The conversations table will allow us to store the previous conversations a user has had with the ai agent.

**Comment by ctron** [comment]
Isn't that personal information that we'd need to be extra careful with? Is that worth the effort?

**Comment by chirino** [comment]
It's expected of a generalized chat UI these days.  Do you have a list of pitfalls that we need to be aware of for this type of data?

**Comment by ctron** [comment]
GDPR, CCPA, and Red Hat policies come into my mind.

**Comment by chirino** [comment]
Those would apply if we are hosting the service right?  Are we hosting the service for the public somewhere?

**Comment by ctron** [comment]
Yes we do. And we shouldn't make it problematic for others to host it too. And at least, give some guidance on what we do, how and why we do it. And what to do if there's no consent. We ourselves also need to give answers to those questions.

**Comment by chirino** [comment]
In the short term, we should recommend not enabling the AI Agent features for public services so that the GRDP, etc issues can be avoided.

**Comment by ctron** [comment]
So we'd need a flag for that, like we do with the graphql. We could enable it by default though.

**Comment by chirino** [comment]
You need to configure an external LLM endpoint and credentials to enable the AI service.  So it's off by default.
